man/systemd-nspawn.xml

```diff
@@ -980,6 +980,20 @@
         effect.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>--notify-mode=</varname></term>
+
+        <listitem><para>Configures support for notifications from init process in the container.
+        <varname>--notify-mode=</varname> supports three options: <option>no</option>, <option>yes</option>, and <option>ready</option>.
+        Option <option>no</option> do not enable notifications between systemd-nspawn and
+        init process in the container and systemd-nspawn notifies systemd with a <literal>READY=1</literal>
+        message when the init process is created. Option <option>yes</option> enables notifications
+        but do nothing with them, systemd-nspawn notifies systemd with a <literal>READY=1</literal>
+        message when the init process is created. Option <option>ready</option>
+        enables notifications and wait for the <literal>READY=1</literal> message from
+        the init process in the container before sending its own to systemd.</para></listitem>
+      </varlistentry>
```

Comment:
well, if we leave it open we could always forward STATUS= for example... After all systemd already can send some STATUS= lines, so it would be good to forward that so that it appears in "systemctl status systemd-nspawn@..." too.

Do you have any specific behaviours in mind for readiness notifications that you'd like to see added? In general, I am very conservative with trying to abstract something and making it generic without knowing what you actually want to cover. If you don't know what precisely to cover, then I'd always say: don't try to abstract it, you can only fail...